src/Core/Grid/Data/Factory/TitleGridDataFactory.php

```diff
@@ -0,0 +1,116 @@
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Grid\Data\Factory;
+
+use PrestaShop\PrestaShop\Adapter\Entity\Gender;
+use PrestaShop\PrestaShop\Core\Grid\Data\GridData;
+use PrestaShop\PrestaShop\Core\Grid\Record\RecordCollection;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+use PrestaShop\PrestaShop\Core\Image\ImageProviderInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * Class TitleGridDataFactory gets data for title grid.
+ */
+final class TitleGridDataFactory implements GridDataFactoryInterface
+{
+    /**
+     * @var GridDataFactoryInterface
+     */
+    private $doctrineTitleDataFactory;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var ImageProviderInterface
+     */
+    private $titleImageThumbnailProvider;
+
+    /**
+     * @param GridDataFactoryInterface $doctrineTitleDataFactory
+     * @param TranslatorInterface $translator
+     * @param ImageProviderInterface $titleImageThumbnailProvider
+     */
+    public function __construct(
+        GridDataFactoryInterface $doctrineTitleDataFactory,
+        TranslatorInterface $translator,
+        ImageProviderInterface $titleImageThumbnailProvider
+    ) {
+        $this->doctrineTitleDataFactory = $doctrineTitleDataFactory;
+        $this->translator = $translator;
+        $this->titleImageThumbnailProvider = $titleImageThumbnailProvider;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData(SearchCriteriaInterface $searchCriteria)
+    {
+        $titleData = $this->doctrineTitleDataFactory->getData($searchCriteria);
+
+        $modifiedRecords = $this->applyModification(
+            $titleData->getRecords()->all()
+        );
+
+        return new GridData(
+            new RecordCollection($modifiedRecords),
+            $titleData->getRecordsTotal(),
+            $titleData->getQuery()
+        );
+    }
+
+    /**
+     * @param array $titles
+     *
+     * @return array
+     */
+    private function applyModification(array $titles): array
+    {
+        foreach ($titles as $i => $title) {
+            switch ($title['type']) {
+                case Gender::GENDER_MALE:
+                    $titles[$i]['type'] = $this->translator->trans('Male', [], 'Admin.Shopparameters.Feature');
+                    break;
+                case Gender::GENDER_FEMALE:
+                    $titles[$i]['type'] = $this->translator->trans('Female', [], 'Admin.Shopparameters.Feature');
+                    break;
+                default:
+                    $titles[$i]['type'] = $this->translator->trans('Neutral', [], 'Admin.Shopparameters.Feature');
+                    break;
```

Comment:
Not sure the default should be `Neutral` 
```suggestion
                case Gender::GENDER_NEUTRAL:
                    $titles[$i]['type'] = $this->translator->trans('Neutral', [], 'Admin.Shopparameters.Feature');
                    break;
                default:
                    throw new BlablablaException(' blabla bla not implemented');
```
It will be easier to customize from the community

Reply:
What is 'blabla bla not implemented'. Do we need wording here? I don't understand what message we want to convey. 🤔

Reply:
@tegbessou Just so you know I'm waiting for an answer here before I can validate the wording